### PR TITLE
fix(server): windows path on linux runtime

### PR DESCRIPTION
**Pull request by alexirion10**

### CODE REVIEW

File: src/SmoFacade/Server.cs [hunk 1]
```diff
@@ -130,24 +142,24 @@ private DefaultFileLocations DefaultFileLocations()
...
+      string result = null;
+      if(!string.IsNullOrWhiteSpace(backupDirectoryQuery)) {
+        using var cmd = SqlConnection.CreateCommand();
+        cmd.CommandText = backupDirectoryQuery;
+        using var reader = cmd.ExecuteReader();
+        if(reader.Read())
+          result = (string)reader[0];
+      }
+
+      result ??= _server.BackupDirectory;
```
Comment (AdityaNPL) [inline src/SmoFacade/Server.cs:156]: Oooo, I didn't know you could do this - `??=`

File: src/SmoFacade/Server.cs [hunk 2]
```diff
@@ -15,81 +15,93 @@ namespace AgDatabaseMove.SmoFacade
...
-                  "AND SUBSTRING(CONVERT(VARCHAR, SERVERPROPERTY('InstanceDefaultDataPath')), 1, 3) = dovs.volume_mount_point"; // Disk Drive equals
-
-      using var cmd = SqlConnection.CreateCommand();
-      cmd.CommandText = query;
-      using var reader = cmd.ExecuteReader();
-      reader.Read();
-      return (int)reader["free_size_mb"];
+      if(Path.IsPathRooted(physicalName))
+        return Path.GetFileName(physicalName) ??
+               throw new InvalidBackupException($"Physical name in backup is incomplete: {physicalName}");
```
Comment (AdityaNPL) [inline src/SmoFacade/Server.cs:99]: What happens if it is not rooted?
Should we throw an exception?

File: src/SmoFacade/Server.cs (hunk 1 above)
Comment (alexirion10) [reply src/SmoFacade/Server.cs:156]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator

File: src/SmoFacade/Server.cs (hunk 2 above)
Comment (alexirion10) [reply src/SmoFacade/Server.cs:99]: See code lines below.... chop off everything before `\\` to get filename

Comment (AdityaNPL) [reply src/SmoFacade/Server.cs:99]: Right for the windows paths, but how bout linux paths?

Comment (alexirion10) [reply src/SmoFacade/Server.cs:99]: I'm pretty sure `IsPathRooted` == true even on windows, but I need to test that more thoroughly

**Pull request merged**